core/src/main/java/org/ldaptive/pool/AbstractConnectionPool.java

```diff
@@ -711,6 +742,59 @@ protected PooledConnectionProxy createConnection(final boolean throwOnFailure)
   }
 
 
+  /**
+   * Asynchronously creates new connections and adds them to the available queue if the connection can be successfully
+   * passivated and validated. See {@link #passivateAndValidateConnection(PooledConnectionProxy)}. This method makes no
+   * guarantees as to the number of connections created only to the number of attempts made.
+   *
+   * @param  count  number of connections to attempt to create
+   * @param  throwOnFailure  whether to throw illegal state exception on any connection creation failure
+   *
+   * @throws  IllegalStateException  if throwOnFailure is true and an exception is thrown attempting to open a
+   *                                 connection
+   */
+  protected void createAvailableConnections(final int count, final boolean throwOnFailure)
+  {
+    poolLock.lock();
+    try {
+      final CallableWorker<PooledConnectionProxy> callableWorker = new CallableWorker<>(getClass().getSimpleName());
+      try {
+        final List<ExecutionException> exceptions = callableWorker.execute(
+          () -> {
+            PooledConnectionProxy pc = createConnection(throwOnFailure);
+            if (pc != null && connectOnCreate) {
+              if (!passivateAndValidateConnection(pc)) {
+                pc.getConnection().close();
+                pc = null;
```

Comment:
There appears to be a behavioral change here. The previous code reads to me like it would try approximately 2x to reach the desired size, but here it appears to be a one-shot attempt. Is that correct and if yes, would you like to implement some equivalent retry logic?

Reply:
Yes, I noted the change in the commit log. I didn't consider alternatives until you commented, but having each `Callable` make two attempts would provide equivalent functionality. I'll push an update for that change.